docs/core/diagnostics/debug-threadpool-starvation.md

```diff
@@ -0,0 +1,280 @@
+    Exception Count (Count / 1 sec)                                0
+    GC Committed Bytes (MB)                                        0
+    GC Fragmentation (%)                                           0
+    GC Heap Size (MB)                                             34
+    Gen 0 GC Count (Count / 1 sec)                                 0
+    Gen 0 Size (B)                                                 0
+    Gen 1 GC Count (Count / 1 sec)                                 0
+    Gen 1 Size (B)                                                 0
+    Gen 2 GC Count (Count / 1 sec)                                 0
+    Gen 2 Size (B)                                                 0
+    IL Bytes Jitted (B)                                      279,021
+    LOH Size (B)                                                   0
+    Monitor Lock Contention Count (Count / 1 sec)                  0
+    Number of Active Timers                                        0
+    Number of Assemblies Loaded                                  121
+    Number of Methods Jitted                                   3,223
+    POH (Pinned Object Heap) Size (B)                              0
+    ThreadPool Completed Work Item Count (Count / 1 sec)           0
+    ThreadPool Queue Length                                        0
+    ThreadPool Thread Count                                        1
+    Time spent in JIT (ms / 1 sec)                                 0.387
+    Working Set (MB)                                              87
+```
+
+The counters above are an example while the web server was not serving any requests. Run Bombadier again with the `api/diagscenario/tasksleepwait` endpoint and sustained load for 2 minutes so there is plenty of time to observe what happens to the performance counters.
+
+```dotnetcli
+>bombardier-windows-amd64.exe https://localhost:5001/api/diagscenario/tasksleepwait -d 120s
+```
+
+Threadpool starvation occurs when there are no free threads to handle the queued workitems and the runtime responds by increasing the number of threadpool threads. You should observe the `ThreadPool Thread Count` rapidly increase to 2-3x the number of processor cores on your machine and then further threads are added 1-2 per second until eventually stabilizing somewhere above 125. The slow and steady increase of threadpool threads combined with CPU Usage much less than 100% are the key signals that threadpool starvation is currently a performance bottleneck. The thread count increase will continue until either the threadpool hits the maximum number of threads, enough threads have been created to satisfy all the incoming workitems, or the CPU has been saturated. Often, but not always, threadpool starvation will also show large values for `ThreadPool Queue Length` and low values for `ThreadPool Completed Work Item Count` showing that there is a large amount of pending work and very little work being completed.
```

Comment:
It would be great to follow up this paragraph with a screen shot of counters that show this pointing out some example numbers.

Reply:
Added a snapshot of dotnet-counters output while the thread count was rising